exonum-java-binding-core/rust/src/utils/errors.rs

```diff
@@ -97,11 +97,14 @@ pub fn get_and_clear_java_exception<'e>(env: &'e JNIEnv) -> JObject<'e> {
 
 fn describe_java_exception(env: &JNIEnv, exception: JObject, jni_error: &JniError) -> String {
     assert!(!exception.is_null(), "No exception thrown.");
-    format!(
-        "Java exception: {}; {:?}\n{}{:#?}",
-        unwrap_jni_verbose(env, get_class_name(env, exception)),
-        unwrap_jni_verbose(env, get_exception_message(env, exception)),
-        unwrap_jni_verbose(env, get_exception_stack_trace(env, exception)),
-        jni_error.backtrace(),
-    )
+    let format = || {
+        Ok(format!(
```

Comment:
I'm against using such implementation on 'happy' paths, where an exception is an expected behaviour (e.g., if a transaction message is malformed, the service *must* throw an exception). I'd restrict usage of this method to actual errors.

On top of that, the exceptions, including errors, are logged on Java side, with proper handling of various stack traces (causes & suppressions).